vcTesting/src/vcTiffTests.cpp

```diff
@@ -47,38 +46,149 @@ static TIFF *vcTesting_LoadTiffTestFile(char const *pFileName)
   return nullptr;
 }
 
-TEST(vcTiff, LoadTiffFile)
+static bool vcTesting_LoadTestFile(char const *pFileName, uint8_t **ppData, int64_t *pSize)
 {
+  for (size_t i = 0; i < udLengthOf(gPossibleFileLocations); ++i)
+  {
+    char buf[64] = {};
+    udSprintf(buf, "%s%s", gPossibleFileLocations[i], pFileName);
+    if (udFile_Load(buf, ppData, pSize) == udR_Success)
+      return true;
+  }
+
+  return false;
+}
+
+TEST(vcTiff, LoadTiffFromMemory)
+{
+  uint8_t *pData = nullptr;
+  int64_t fileSize = 0;
+  vcMemoryStream *pStream = nullptr;
   TIFF *pTif = nullptr;
   uint32 w = 0;
   uint32 h = 0;
   size_t npixels = 0;
   int directoryCount = 0;
   uint32 *raster = nullptr;
+  udResult result;
+  int intRes;
 
   do
   {
-    pTif = vcTesting_LoadTiffTestFile(TEST_FILE_1_NAME);
+    bool loaded  = vcTesting_LoadTestFile(TEST_FILE_1_NAME, &pData, &fileSize);
+    EXPECT_TRUE(loaded);
+    if (!loaded)
+      break;
+
+    result = vcTiffHandler_CreateMemoryInStream(pData, size_t(fileSize), &pStream);
+    EXPECT_EQ(result, udR_Success);
+    if (result != udR_Success)
+      break;
 
-    EXPECT_NE_BREAK(pTif, nullptr);
+    result= vcTiffHandler_CreateTiffInStream(pStream, &pTif);
+    EXPECT_EQ(result, udR_Success);
+    if (result != udR_Success)
+      break;
 
     // Directories
     directoryCount = vcTesting_TiffDirectoryCount(pTif);
-    EXPECT_EQ_BREAK(directoryCount, 1);
+    EXPECT_EQ(directoryCount, 1);
+    if (directoryCount != 1)
+      break;
 
     // Colour data
-    EXPECT_EQ_BREAK(TIFFGetField(pTif, TIFFTAG_IMAGEWIDTH, &w), 1);
-    EXPECT_EQ_BREAK(w, 3ul);
-    EXPECT_EQ_BREAK(TIFFGetField(pTif, TIFFTAG_IMAGELENGTH, &h), 1);
-    EXPECT_EQ_BREAK(h, 3ul);
+    intRes = TIFFGetField(pTif, TIFFTAG_IMAGEWIDTH, &w);
+    EXPECT_EQ(intRes, 1);
+    if (intRes != 1)
+      break;
+
+    EXPECT_EQ(w, 3ul);
+    if (w != 3ul)
+      break;
+
+    intRes = TIFFGetField(pTif, TIFFTAG_IMAGELENGTH, &h);
+    EXPECT_EQ(intRes, 1);
+    if (intRes != 1)
+      break;
+
+    EXPECT_EQ(h, 3ul);
+    if (h != 3ul)
+      break;
 
     npixels = size_t(w) * h;
 
     raster = (uint32 *)_TIFFmalloc(npixels * sizeof(uint32));
-    EXPECT_NE_BREAK(raster, nullptr);
+    EXPECT_NE(raster, nullptr);
+    if (raster == nullptr)
+      break;
+
+    intRes = TIFFReadRGBAImageOriented(pTif, w, h, raster, 0, ORIENTATION_TOPLEFT);
+    EXPECT_EQ(intRes, 1);
+    if (intRes != 1)
+      break;
 
-    int readSuccess = TIFFReadRGBAImageOriented(pTif, w, h, raster, 0, ORIENTATION_TOPLEFT);
-    EXPECT_EQ_BREAK(readSuccess, 1);
+    EXPECT_EQ(memcmp(gTEST_FILE_1_PIXELS, raster, npixels * sizeof(uint32)), 0);
+  } while (false);
+
+  _TIFFfree(raster);
+  TIFFClose(pTif);
+  vcTiffHandler_DestroyMemoryStream(&pStream);
+  udFree(pData);
+}
+
+TEST(vcTiff, LoadTiffFile)
```

Comment:
Turns out we don't actually use libtiff to load tiff from file. But this test exists nonetheless.